pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -806,25 +867,33 @@ type UnschedulablePods struct {
 	// podInfoMap is a map key by a pod's full-name and the value is a pointer to the QueuedPodInfo.
 	podInfoMap map[string]*framework.QueuedPodInfo
 	keyFunc    func(*v1.Pod) string
-	// metricRecorder updates the counter when elements of an unschedulablePodsMap
-	// get added or removed, and it does nothing if it's nil
-	metricRecorder metrics.MetricRecorder
+	// metricRecorder/gatedRecorder updates the counter when elements of an unschedulablePodsMap
```

Comment:
No metricRecorder now

Reply:
And this seems like it will impact current metrics now.

Reply:
> No metricRecorder now

updated.

> And it seems like will impact current metrics now.

Metric's sub-label addition should be ok I suppose, as only users adopting the new feature will use the metric with new label; otherwise, all unschedulable pods will still fall into the metric with old labels.